src/gondi.ts

```diff
@@ -179,9 +193,16 @@ export class Gondi {
       tokenId,
       taker,
     };
-    const typedData = await this.api.generateNftOrderToBeSigned(orderInput);
-    const signature = await this.wallet.signTypedData(typedData);
-    return this.api.saveSignedNftOrder({ ...orderInput, signature });
+    let response = await this.api.publishOrder(orderInput);
+    while (response.__typename === 'SignatureRequest') {
```

Comment:
I'd assume the order we expect here. I wouldn't iterate indefinitely.

Reply:
The number of signature depends if it's ask or not